src/Components/ResultsFilter.tsx

```diff
@@ -1,25 +1,43 @@
+import { useState } from 'react';
 import ResultsFilterItem from '../Components/ResultsFilterItem';
 
-const filterOptions = [
-  {
-    id: 1,
-    text: 'All',
-    active: true
-  },
-  { id: 2, text: 'Documentary', active: false },
-  { id: 3, text: 'Comedy', active: false },
-  { id: 4, text: 'Horror', active: false },
-  { id: 5, text: 'Crime', active: false }
-];
-
 export default function ResultsFilter() {
+  const [filterOptions, setFilterOptions] = useState([
+    {
+      id: 1,
+      text: 'All',
+      active: true
+    },
+    { id: 2, text: 'Documentary', active: false },
+    { id: 3, text: 'Comedy', active: false },
+    { id: 4, text: 'Horror', active: false },
+    { id: 5, text: 'Crime', active: false }
+  ]);
+
+  const clickCategoryHandler = (id: number) => {
```

Comment:
This could have been simplified by keeping active as a separate state

Reply:
sure I can look into it next time